### PR TITLE
Add new plan form

### PR DESCRIPTION
**Pull request by abdulwahidsharief**
### - Link added should be a valid URL and accessible throughout the org.
   ### - If the branch name contains hotfix / revert by default the BVT workflow check will pass.

| Test Case Document URL                        |
|-----------------------------------------------|
| Please paste test case document link here.... |

### CODE REVIEW

File: admin/controller/payment/razorpay.php [hunk 1]
```diff
@@ -311,6 +311,466 @@ public function uninstall(): void {
...
+            $data['entity_id'] = 0;
+        }
+
+        if (isset($this->error['warning']))
+        {
+            $data['error_warning'] = $this->error['warning'];
+        }
+        else
+        {
+            $data['error_warning'] = '';
```
Comment (ramth05) [inline admin/controller/payment/razorpay.php:333]: @abdulwahidsharief , assign all empty variables at the start. no need to use else block.

File: admin/view/template/payment/razorpay_subscription/razorpay_plan_form.twig [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+			</div>	
+		</div>
+	</div>
+</div>
+<script type="text/javascript"><!--
+//auto load product id
+$('#input-product').autocomplete({
+	'source': function(request, response) {
+		$.ajax({
+			url: 'index.php?route=catalog/product.autocomplete&user_token={{ user_token }}&filter_name=' +  encodeURIComponent(request),
```
Comment (ramth05) [inline admin/view/template/payment/razorpay_subscription/razorpay_plan_form.twig:166]: @abdulwahidsharief , is there any method to form the url? this looks like hardcoded and any changes in product may break it.

File: admin/model/payment/razorpay.php [hunk 3]
```diff
@@ -496,54 +496,53 @@ public function getSingleSubscription($entity_id)
...
     {
-        $this->rzpPdo->prepare("INSERT INTO `" . DB_PREFIX . "recurring` SET `status` = :status, `price` = :price, `frequency` = :frequency, `duration` = :duration, `cycle` = :cycle, `trial_status` = :trial_status, `trial_price` = :trial_price, `trial_frequency` = :trial_frequency, `trial_duration` = :trial_duration, `trial_cycle` = :trial_cycle");
-        $this->rzpPdo->bindParam(':status', (int)$data['status']);
-        $this->rzpPdo->bindParam(':price', (float)$data['price']);
-        $this->rzpPdo->bindParam(':frequency', $this->db->escape($data['frequency']));
-        $this->rzpPdo->bindParam(':duration', (int)$data['duration']);
-        $this->rzpPdo->bindParam(':cycle', (int)$data['cycle']);
-        $this->rzpPdo->bindParam(':trial_status', (int)$data['trial_status']);
-        $this->rzpPdo->bindParam(':trial_price', (float)$data['trial_price']);
+        $this->rzpPdo->prepare("INSERT INTO `" . DB_PREFIX . "subscription_plan` SET `trial_frequency` = :trial_frequency, `trial_duration` = :trial_duration, `trial_cycle` = :trial_cycle, `trial_status` = :trial_status, `frequency` = :frequency, `duration` = :duration, `cycle` = :cycle, `status` = :status");
```
Comment (ramth05) [inline admin/model/payment/razorpay.php:501]: @abdulwahidsharief , please create new variable for query and use proper line for each columns.

File: admin/view/template/payment/razorpay_subscription/razorpay_plan_form.twig (hunk 2 above)
Comment (abdulwahidsharief) [reply admin/view/template/payment/razorpay_subscription/razorpay_plan_form.twig:166]: @ramth05 didn't find any method to get product catalog url

Comment (abdulwahidsharief) [reply admin/view/template/payment/razorpay_subscription/razorpay_plan_form.twig]: changed to catalog_product_url

**Review by ramth05 — APPROVED** [review]
LGTM, please test it after master merge thoroughly

**Pull request merged**